pkg/operator/oauthapiencryptioncontroller/oauthapiencryptioncontroller.go

```diff
@@ -0,0 +1,129 @@
+// case 4: no-op: when the secret doesn't exist and encryption is off
+//
+// drawbacks:
+// - it will not recover when the annotation was manually removed by a user,
+//   to recover we would have to put a value in the annotation instead and coordinate OAS-A and CAO but then we would have to remember to add it in CAO (4.6) as well
+func (c *oauthAPIServerController) sync(ctx context.Context, controllerContext factory.SyncContext) error {
+	openshiftAPIServerEncryptionCfg, err := c.secretLister.Get(fmt.Sprintf("%s-%s", encryptionconfig.EncryptionConfSecretName, c.targetNamespace))
+	if apierrors.IsNotFound(err) {
+		return nil // case 4: encryption off
+	}
+	if err != nil {
+		return err
+	}
+
+	oauthAPIServerEncryptionCfgName := fmt.Sprintf("%s-%s", encryptionconfig.EncryptionConfSecretName, c.oauthAPIServerTargetNamespace)
+	oauthAPIServerEncryptionCfg, err := c.secretLister.Get(oauthAPIServerEncryptionCfgName)
+	if err != nil && !apierrors.IsNotFound(err) {
+		return err
+	}
+	if apierrors.IsNotFound(err) {
+		// case 1: create with annotation
+		if _, exists := openshiftAPIServerEncryptionCfg.Data[encryptionconfig.EncryptionConfSecretKey]; !exists {
+			return fmt.Errorf("%s/%s doesn't contain the required key %q", openshiftAPIServerEncryptionCfg.Name, openshiftAPIServerEncryptionCfg.Namespace, encryptionconfig.EncryptionConfSecretKey)
+		}
+		encryptionCfg := &corev1.Secret{}
+		encryptionCfg.Name = oauthAPIServerEncryptionCfgName
+		encryptionCfg.Namespace = operatorclient.GlobalMachineSpecifiedConfigNamespace
+		encryptionCfg.Annotations = map[string]string{}
+		encryptionCfg.Annotations[encryptionConfigManagedBy] = encryptionConfigManagedByValue
+		encryptionCfg.Annotations[encryptionstate.KubernetesDescriptionKey] = encryptionstate.KubernetesDescriptionScaryValue
+		encryptionCfg.Data = map[string][]byte{}
+		encryptionCfg.Data[encryptionconfig.EncryptionConfSecretKey] = openshiftAPIServerEncryptionCfg.Data[encryptionconfig.EncryptionConfSecretKey]
+		encryptionCfg.Finalizers = []string{encryptionsecret.EncryptionSecretFinalizer}
+
+		_, err := c.secretClient.Create(ctx, encryptionCfg, metav1.CreateOptions{})
+		if err != nil {
+			return err
+		}
+		controllerContext.Recorder().Eventf("SecretCreated", "Created %s in %s namespace because it was missing", oauthAPIServerEncryptionCfgName, operatorclient.GlobalMachineSpecifiedConfigNamespace)
+		return nil
+	}
+	if _, exist := oauthAPIServerEncryptionCfg.Annotations[encryptionConfigManagedBy]; exist {
+		// case 2: exists and it is annotated
+		oauthEncryptionCfgData := oauthAPIServerEncryptionCfg.Data[encryptionconfig.EncryptionConfSecretKey]
+		oasEncryptionCfgData := openshiftAPIServerEncryptionCfg.Data[encryptionconfig.EncryptionConfSecretKey]
+		if !equality.Semantic.DeepEqual(oauthEncryptionCfgData, oasEncryptionCfgData) {
+			encryptionCfg := oauthAPIServerEncryptionCfg.DeepCopy()
+			encryptionCfg.Data[encryptionconfig.EncryptionConfSecretKey] = oasEncryptionCfgData
```

Comment:
didn't we have any annotations that play a role? Or just for keys?

Reply:
we actually use https://github.com/openshift/library-go/blob/master/pkg/operator/encryption/encryptionconfig/secret.go#L71 but the content is the same :) (uff)